common/VotingUtils.js

```diff
@@ -17,8 +22,15 @@ const constructCommitment = async (request, roundId, web3, price, account) => {
   const hash = web3.utils.soliditySha3(priceWei, salt);
 
   const vote = { price: priceWei, salt };
-  const { publicKey } = await deriveKeyPairFromSignatureTruffle(web3, getKeyGenMessage(roundId), account);
-  const encryptedVote = await encryptMessage(publicKey, JSON.stringify(vote));
+  let derivedPublicKey;
+  if (web3.currentProvider.label === "metamask") {
```

Comment:
nit: this is kinda nonstandard. Why not grab the `--network` arg like we do here: https://github.com/UMAprotocol/protocol/blob/904e0454b1bc09903add91fe0b562f32c7de0faf/common/MetaMaskTruffleProvider.js#L48.

You'll have to initialize minimist like we do here: https://github.com/UMAprotocol/protocol/blob/904e0454b1bc09903add91fe0b562f32c7de0faf/common/MetaMaskTruffleProvider.js#L2

Reply:
Much better this way thanks!

Reply:
done